### ISSUE TITLE
'gpio_chip' has no member named 'dev' 

### ISSUE THREAD

**Issue by mrnamwen** [marcusbirkin/BGT3xxx#10]
I'm trying to compile this driver for my BGT3600 card. It appears that on both DKMS and standard compile that this specific error keeps occurring.

Distro is Arch Linux (latest kernel release - 4.10). All kernel headers are installed.

DKMS Log:
```DKMS make.log for BGT3xxx-1.0.0.0Fixed for kernel 4.10.6-1-ARCH (x86_64)
Sun  2 Apr 16:26:28 BST 2017
make -C /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l 
make[1]: Entering directory '/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l'
scripts/make_makefile.pl
./scripts/make_kconfig.pl /lib/modules/4.10.6-1-ARCH/build /lib/modules/4.10.6-1-ARCH/build
Updating/Creating .config
Preparing to compile for kernel version 4.10.6

***WARNING:*** You do not have the full kernel sources installed.
This does not prevent you from building the v4l-dvb tree if you have the
kernel headers, but the full kernel source may be required in order to use
make menuconfig / xconfig / qconfig.

If you are experiencing problems building the v4l-dvb tree, please try
building against a vanilla kernel before reporting a bug.

Vanilla kernels are available at http://kernel.org.
On most distros, this will compile a newly downloaded kernel:

cp /boot/config-`uname -r` <your kernel dir>/.config
cd <your kernel dir>
make all modules_install install

Please see your distro's web site for instructions to build a new kernel.

./scripts/make_myconfig.pl
perl scripts/make_config_compat.pl /lib/modules/4.10.6-1-ARCH/build ./.myconfig ./config-compat.h
ln -sf . oss
make -C firmware prep
creating symbolic links...
make[2]: Entering directory '/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/firmware'
make[2]: Leaving directory '/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/firmware'
make -C firmware
make[2]: Entering directory '/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/firmware'
  CC  ihex2fw
Generating vicam/firmware.fw
Generating dabusb/firmware.fw
Generating dabusb/bitstream.bin
Generating ttusb-budget/dspbootcode.bin
Generating cpia2/stv0672_vp4.bin
Generating av7110/bootcode.bin
make[2]: Leaving directory '/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/firmware'
Kernel build directory is /lib/modules/4.10.6-1-ARCH/build
make -C /lib/modules/4.10.6-1-ARCH/build SUBDIRS=/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l CFLAGS="-I/usr/include -D__KERNEL__ -I/include -DEXPORT_SYMTAB" modules
make[2]: Entering directory '/usr/lib/modules/4.10.6-1-ARCH/build'
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/tuner-simple.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/tuner-types.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/tda18271-maps.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/tda18271-common.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/tda18271-fe.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvbdev.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dmxdev.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_demux.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_filter.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_ca_en50221.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_frontend.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_net.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_ringbuffer.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_math.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb-pll.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/tda18272.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/tda10048.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/s5h1411.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/dvb_dummy_fe.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/stv090x.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/stv6110x.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/lnbp21.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2820r_core.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2820r_c.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2820r_t.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2820r_t2.o
  CC [M]  /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2850.o
/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2820r_core.c: In function 'cxd2820r_attach':
/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2820r_core.c:739:18: error: 'struct gpio_chip' has no member named 'dev'
   priv->gpio_chip.dev = &priv->i2c->dev;
                  ^
make[3]: *** [scripts/Makefile.build:295: /var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l/cxd2820r_core.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [Makefile:1490: _module_/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l] Error 2
make[2]: Leaving directory '/usr/lib/modules/4.10.6-1-ARCH/build'
make[1]: *** [Makefile:44: default] Error 2
make[1]: Leaving directory '/var/lib/dkms/BGT3xxx/1.0.0.0Fixed/build/v4l'
make: *** [Makefile:27: all] Error 2```

**Comment by marcusbirkin** [comment]
This would appear to be a change in Linux kernel >=4.5, I will sort out a patch this week.
In the mean time, you could use Linux kernel 4.4 or older.

**Comment by mrnamwen** [comment]
Great, thanks. I'd build the patch myself but my experience is in higher
level languages, rather then low level ones like C and C++. I'll try out
4.4 in the meantime.

**Comment by marcusbirkin** [comment]
Fixed in commit 7fef4ceaa60fd88301115dfbcf961bda6245695f
